test 'tsickle compiler host passes blacklisted paths' fails on Windows

Last remaining test that fails on Windows:

```
1) tsickle compiler host passes blacklisted paths:
   AssertionError: expected 'let /** @type {!Banned} */ b: Banned = {b: "a"};' to include '/** @type {?} */ b: Banned'
   at Context.<anonymous> (D:\workspace\b-strauss\test\e2e_tsickle_compiler_host_test.ts:83:23)
```

--- a/src/type-translator.ts
+++ b/src/type-translator.ts
@@ -511,13 +511,14 @@ export class TypeTranslator {
   /** @return true if sym should always have type {?}. */
   isBlackListed(symbol: ts.Symbol): boolean {
     if (this.pathBlackList === undefined) return false;
+    const pathBlackList = this.pathBlackList;
     if (symbol.declarations === undefined) {
       this.warn('symbol has no declarations');
       return true;
     }
     return symbol.declarations.every(n => {
       const fileName = path.normalize(n.getSourceFile().fileName);
-      return this.pathBlackList != null && this.pathBlackList.has(fileName);
+      return pathBlackList.has(fileName);
     });
   }
 }

--- a/src/type-translator.ts
+++ b/src/type-translator.ts
@@ -6,6 +6,7 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import * as nodePath from 'path';
 import * as ts from 'typescript';
 
 export function assertTypeChecked(sourceFile: ts.SourceFile) {
@@ -503,13 +504,15 @@ export class TypeTranslator {
   /** @return true if sym should always have type {?}. */
   isBlackListed(symbol: ts.Symbol): boolean {
     if (this.pathBlackList === undefined) return false;
-    const pathBlackList = this.pathBlackList;
     if (symbol.declarations === undefined) {
       this.warn('symbol has no declarations');
       return true;
     }
+    // Normalize paths to not break checks on Windows.
+    const pathBlackList = new Set<string>(
+        Array.from(this.pathBlackList.values()).map(path => nodePath.normalize(path)));
     return symbol.declarations.every(n => {
-      const path = n.getSourceFile().fileName;
+      const path = nodePath.normalize(n.getSourceFile().fileName);
       return pathBlackList.has(path);
     });
   }

--- a/src/type-translator.ts
+++ b/src/type-translator.ts
@@ -6,7 +6,7 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import * as nodePath from 'path';
+import * as path from 'path';
 import * as ts from 'typescript';
 
 export function assertTypeChecked(sourceFile: ts.SourceFile) {
@@ -135,7 +135,7 @@ export class TypeTranslator {
    * A list of types we've encountered while emitting; used to avoid getting stuck in recursive
    * types.
    */
-  private seenTypes: ts.Type[] = [];
+  private readonly seenTypes: ts.Type[] = [];
 
   /**
    * @param node is the source AST ts.Node the type comes from.  This is used
@@ -147,9 +147,16 @@ export class TypeTranslator {
    *     (`tsickle_import.Foo`).
    */
   constructor(
-      private typeChecker: ts.TypeChecker, private node: ts.Node,
-      private pathBlackList?: Set<string>,
-      private symbolsToAliasedNames: Map<ts.Symbol, string> = new Map<ts.Symbol, string>()) {}
+      private readonly typeChecker: ts.TypeChecker, private readonly node: ts.Node,
+      private readonly pathBlackList?: Set<string>,
+      private readonly symbolsToAliasedNames:
+          Map<ts.Symbol, string> = new Map<ts.Symbol, string>()) {
+    // Normalize paths to not break checks on Windows.
+    if (this.pathBlackList != null) {
+      this.pathBlackList =
+          new Set<string>(Array.from(this.pathBlackList.values()).map(p => path.normalize(p)));
+    }
+  }
 
   /**
    * Converts a ts.Symbol to a string.
@@ -508,12 +515,9 @@ export class TypeTranslator {
       this.warn('symbol has no declarations');
       return true;
     }
-    // Normalize paths to not break checks on Windows.
-    const pathBlackList = new Set<string>(
-        Array.from(this.pathBlackList.values()).map(path => nodePath.normalize(path)));
     return symbol.declarations.every(n => {
-      const path = nodePath.normalize(n.getSourceFile().fileName);
-      return pathBlackList.has(path);
+      const fileName = path.normalize(n.getSourceFile().fileName);
+      return this.pathBlackList != null && this.pathBlackList.has(fileName);
     });
   }
 }

